applications/MappingApplication/custom_mappers/mapper.h

```diff
@@ -102,44 +72,91 @@ class Mapper
+    * @param SearchRadius search radius used for the search
+    */
+    virtual void UpdateInterface(
+        Kratos::Flags MappingOptions,
+        double SearchRadius) = 0;
+
+    /**
+    * @brief Mapping from Origin to Destination, Scalar Variable
+    * Data is exchanged on the Interface, from the Origin-Modelpart
+    * to the Destination-ModelPart (the modelparts were specified in the
+    * construction Phase of the Mapper)
+    * @param rOriginVariable Variable on the Origin-ModelPart
+    * @param rDestinationVariable Variable on the Destination-ModelPart
+    * @param MappingOptions flags used to specify options for the mapping
+    * @see InverseMap
+    */
+    virtual void Map(
+        const Variable<double>& rOriginVariable,
+        const Variable<double>& rDestinationVariable,
+        Kratos::Flags MappingOptions) = 0;
+
+    /**
+    * @brief Mapping from Origin to Destination, Vector Variable
+    * Same as Map, but maps an array3-variable
+    * @see Map
+    */
+    virtual void Map(
+        const Variable< array_1d<double, 3> >& rOriginVariable,
+        const Variable< array_1d<double, 3> >& rDestinationVariable,
+        Kratos::Flags MappingOptions) = 0;
+
+    /**
+    * @brief Mapping from Destination to Origin, Scalar Variable
+    * Data is exchanged on the Interface, from the Destination-Modelpart
+    * to the Origin-ModelPart (the modelparts were specified in the
+    * construction Phase of the Mapper)
+    * It does the opposite of Map
+    * @param rOriginVariable Variable on the Origin-ModelPart
+    * @param rDestinationVariable Variable on the Destination-ModelPart
+    * @param MappingOptions flags used to specify options for the mapping
+    * @see Map
+    */
+    virtual void InverseMap(
+        const Variable<double>& rOriginVariable,
+        const Variable<double>& rDestinationVariable,
+        Kratos::Flags MappingOptions) = 0;
+
+    /**
+    * @brief Mapping from Destination to Origin, Vector Variable
+    * Same as InveseMap, but maps an array3-variable
+    * @see InverseMap
+    */
+    virtual void InverseMap(
+        const Variable< array_1d<double, 3> >& rOriginVariable,
+        const Variable< array_1d<double, 3> >& rDestinationVariable,
+        Kratos::Flags MappingOptions) = 0;
+
+    /**
+    * @brief Cloning the Mapper
+    * returns a clone of the current Mapper
+    * pure virtual, has to be implemented in every derived mapper,
+    * used in the creation of the Mappers
+    * @see MapperFactory
+    */
+    virtual MapperUniquePointerType Clone(
+        ModelPart& rModelPartOrigin,
+        ModelPart& rModelPartDestination,
+        Parameters JsonParameters) const = 0;
 
     ///@}
     ///@name Access
     ///@{
 
-    ///@}
-    ///@name Inquiry
-    ///@{
+    /**
+     * @brief This method returns the mapping-matrix
+     * @return The mapping-matrix
+     */
+    virtual TMappingMatrixType& GetMappingMatrix() = 0;
```

Comment:
Since the return value is optional, you could return a ptr instead of a ref. Implementations that don't store a matrix can return nullptr rather than throwing.
```suggestion
    virtual TMappingMatrixType* pGetMappingMatrix() = 0;
```